version/src/lib.rs

```diff
@@ -19,66 +21,81 @@ use serde::{Deserialize, Serialize};
-        #[error("This is not a versioned object. No version information found.")]
+        /// This is not a versioned object
         NotVersioned,
-        /// Can not encode unsupported version from json to scale.
-        #[error("Can not encode unsupported version from json to scale.")]
+        /// Cannot encode unsupported version from JSON to Parity SCALE
         UnsupportedJsonEncode,
-        /// Expected json object.
-        #[error("Expected json object.")]
+        /// Expected JSON object
         ExpectedJson,
-        /// Can not encode unsupported version from scale to json
-        #[error("Can not encode unsupported version from scale to json.")]
+        /// Cannot encode unsupported version from Parity SCALE to JSON
         UnsupportedScaleEncode,
-        /// Problem with serialization/deserialization of json
-        #[error("Problem with serialization/deserialization of json.")]
-        Serde(#[source] serde_json::Error),
-        /// Problem with serialization/deserialization of parity scale.
-        #[error("Problem with serialization/deserialization of parity scale.")]
-        ParityScale(#[source] parity_scale_codec::Error),
-        /// Problem with parsing integers.
-        #[error("Problem with parsing integers.")]
-        ParseInt(#[source] std::num::ParseIntError),
-        /// Version of input is unsupported
-        #[error("Version of input is unsupported")]
+        /// JSON (de)serialization issue
+        #[cfg(feature = "json")]
+        Serde(#[cfg_attr(feature = "std", source)] serde_json::Error),
+        /// Parity SCALE (de)serialization issue
+        #[cfg(feature = "scale")]
+        ParityScale(#[cfg_attr(feature = "std", source)] parity_scale_codec::Error),
+        /// Problem with parsing integers
+        ParseInt(#[cfg_attr(feature = "std", source)] core::num::ParseIntError),
+        /// Input version unsupported
         UnsupportedVersion(UnsupportedVersion),
     }
 
+    impl fmt::Display for Error {
+        fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+            let msg = match *self {
+                Self::NotVersioned => "Not a versioned object",
+                Self::UnsupportedJsonEncode => {
+                    "Cannot encode unsupported version from JSON to SCALE"
+                }
+                Self::ExpectedJson => "Expected JSON object",
+                Self::UnsupportedScaleEncode => {
+                    "Cannot encode unsupported version from SCALE to JSON"
+                }
+                #[cfg(feature = "json")]
+                Self::Serde(_) => "JSON (de)serialization issue",
+                #[cfg(feature = "scale")]
+                Self::ParityScale(_) => "Parity SCALE (de)serialization issue",
+                Self::ParseInt(_) => "Problem with parsing integers",
+                Self::UnsupportedVersion(_) => "Input version unsupported",
+            };
+
+            write!(f, "{}", msg)
+        }
+    }
```

Comment:
Maybe we can use [`derive_more`](https://docs.rs/derive_more/latest/derive_more/index.html), which supports `no_std` ?

Reply:
Idk what to say. I considered it but decided against it. It doesn't seem to bring much to the table for these small crates.
I can add `derive_more::Display` if you think it would bring something?

Reply:
I'd like to accept your decision then